purchase_sale_inter_company/models/purchase_order.py

```diff
@@ -3,26 +3,76 @@
 # Copyright 2018-2019 Tecnativa - Carlos Dauden
 # License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
 
-from odoo import _, models
+from odoo import _, fields, models
 from odoo.exceptions import UserError
 
 
 class PurchaseOrder(models.Model):
     _inherit = "purchase.order"
 
+    auto_sale_order_info = fields.Html(compute="_compute_auto_sale_order_info")
+
+    def _compute_auto_sale_order_info(self):
+        for purchase_order in self.sudo():
+            auto_sale_order_id = (
+                self.env["sale.order"]
+                .sudo()
+                .search([("auto_purchase_order_id", "=", purchase_order.id)])
+            )
+            info_message = (
+                _(
+                    "Sale order '%s' is already generated for this purchase order "
+                    "You can confirm/cancel this purchase order and "
+                    "related sale order will be also confirmed/canceled"
+                )
+                % auto_sale_order_id.display_name
+            )
+            purchase_order.auto_sale_order_info = (
+                info_message if auto_sale_order_id.exists() else False
+            )
+
     def button_approve(self, force=False):
         """Generate inter company sale order base on conditions."""
         res = super().button_approve(force)
         for purchase_order in self.sudo():
             # get the company from partner then trigger action of
             # intercompany relation
+            po_partner = purchase_order.partner_id
+            # vendor can be a member of group company or it's child
+            # dest_company will be anyway member of group company
             dest_company = (
-                purchase_order.partner_id.commercial_partner_id.ref_company_ids
+                po_partner.ref_company_ids or po_partner.parent_id.ref_company_ids
             )
+            intercompany_user = dest_company.intercompany_sale_user_id
+            if not intercompany_user:
+                intercompany_user = self.env.user
```

Comment:
These lines may return a `res.users` recordset as `intercompany_user`, but at line 70 we have `intercompany_user.id` which will raise an error if there's more than 1 record. Please fix this.